cpp/lib/include/Subfields.h

```diff
@@ -21,46 +21,71 @@
 #define SUBFIELDS_H
 
 
+#include <algorithm>
+#include <iostream>
+#include <PriorityQueue.h>
 #include <unordered_map>
 #include <string>
 #include <vector>
 
 
+struct Subfield {
+    char code_;
+    std::string value_;
+public:
+    Subfield(const char code, const std::string &value) : code_(code), value_(value) {}
+};
+
+struct SubfieldCodeLessThan {
+    bool operator() (const Subfield &lhs, const Subfield &rhs) { return lhs.code_ < rhs.code_; }
```

Comment:
I think these operator functions can be const.  Making them const allows to use them more broadly and may enable the compiler to use additional optimisations.